pkg/modulewriter/modulewriter.go

```diff
@@ -133,38 +127,92 @@ func createGroupDirs(deploymentPath string, deploymentGroups *[]config.Deploymen
 	return nil
 }
 
-func copySource(deploymentPath string, deploymentGroups *[]config.DeploymentGroup) error {
+// Get module source within deployment group
+// Rules are following:
+//   - git source
+//     => keep the same source
+//   - packer
+//     => <mod.ID>
+//   - embedded (source starts with "modules" or "comunity/modules")
+//     => ./modules/embedded/<source>
+//   - other
+//     => ./modules/<basename(source)>-<hash(abs(source))>
+func deploymentSource(mod config.Module) (string, error) {
+	if sourcereader.IsGitPath(mod.Source) {
+		return mod.Source, nil
+	}
+	switch mod.Kind {
+	case "packer":
+		return mod.ID, nil
+	case "terraform": // see below
+	default:
+		return "", fmt.Errorf("unexpected module kind %#v", mod.Kind)
+	}
```

Comment:
nit: 😬 I am not a huge fan of this control flow. It strikes me as a bit clever use a switch for return early with a fall through. You also are mixing patterns as you have several if return early and then the switch. My preference would be for two more return early if blocks like you have for the isGitPath.

```
	if sourcereader.IsGitPath(mod.Source) {
		return mod.Source, nil
	}
	if mod.Kind == "packer" {
		return mod.ID, nil
	}
	if mod.Kind != "terraform" {
		return "", fmt.Errorf("unexpected module kind %#v", mod.Kind)
	}
...
```